packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -527,9 +535,10 @@ class HotRunner extends ResidentRunner {
   @override
   Future<OperationResult> restart({
     bool fullRestart = false,
-    bool pauseAfterRestart = false,
     String reason,
     bool benchmarkMode = false,
+    bool silent = false,
+    bool pause = false, // this is not currently supported for hot restart.
```

Comment:
Should this comment say "fast start" and not "hot restart"?

Reply:
Its not supported for hot restart confusingly, only hot reload. I've updated the parameter name to better match the usecase

Reply:
Now I'm confused :) Do you just mean when using `--fast-start`? The [daemon docs](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/doc/daemon.md#apprestart) specifically call this out for hot restart:

> `pause`: optional; when doing a hot restart the isolate should enter a paused mode

(though tbh, I'm sure in VS Code we assume that it works fine for both reload and restart... if it doesn't I'm surprised we're not seeing lots of weird issues since it's when we get pause events we re-send the breakpoints)

Reply:
I haven't noticed that working ever ... let me get back to you on that

Reply:
I had a quick test, and it does seem to pause. Here I sent a hot restart with `pause:true` and the isolate went `IsolateRunnable`->`PauseStart`, then I sent breakpoints and `resume`'d and it continued:

<img width="1520" alt="Screenshot 2019-11-28 at 9 19 15 am" src="https://user-images.githubusercontent.com/1078012/69793486-71de2b80-11c0-11ea-9e24-e4fe451da2e3.png">

Reply:
This is great news! Just need to figure out what I was doing wrong...  Thanks @DanTup !

Reply:
TLDR: --start-paused, applies to all isolates, not just the first spawned.

Reply:
Ah yeah, in the IDEs we automatically resume any isolates that appear in the `PauseStart` state as we get events about them (even well after the app has started).